controllers/nnf_node_slc.go

```diff
@@ -156,18 +152,31 @@ func (r *NnfNodeSLCReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	switch storage.Spec.State {
 	case dwsv1alpha1.EnabledState:
+
 		// Clear the fenced status if the node is enabled from a disabled status
 		if storage.Status.Status == dwsv1alpha1.DisabledStatus {
-			nnfNode.Status.Fenced = false
 
-			storage.Status.RebootRequired = false
-			storage.Status.Message = ""
+			if nnfNode.Status.Fenced {
+				log.Info("Clearing fencing status")
+
+				nnfNode.Status.Fenced = false
+
+				if err := r.Status().Update(ctx, nnfNode); err != nil {
```

Comment:
We talked about this over slack, but I'm having second thoughts now. In our other reconcilers, the defer status updater is for the resource that is being reconciled. That's the NnfNode in this case. The Storage resource is the weird case where we have a defer status updater, but it's not the resource being reconciled.